### ISSUE TITLE
PR #1556 makes new data not appear

### ISSUE THREAD

**Issue by EfremBraun** [pyqtgraph/pyqtgraph#1676]
PR #1556 introduced a flaw where when new data is plotted, the data might not show up until the user pans the scene (if that data wouldn't have shown up prior to the fix given in PR #1140 ). This is unfortunately my use case, as I allow my user to show/hide plot elements.

I'm not sure exactly how to fix this, but it looks like PR #1556 made it so that the  fix done in PR #1140 is only performed upon panning.

A minimal working example is below. It's copy/pasted from plot_test_5.zip given in PR #1556 with a button added to change the data that is being shown. If you run this example using any version of PyQtGraph past commit 33365fb (PR #1556), upon hitting the button you'll see see the magenta curve disappear. If you run it with a version of PyQtGraph prior to commit 33365fb but after commit 65e90fa (PR #1440), the magenta curve will not disappear.

```
from PySide2 import QtWidgets
import sys
import numpy as np
import pyqtgraph as pg
# import dev_pyqtgraph.pyqtgraph as pg

class MainWindow(QtWidgets.QMainWindow):
    def __init__(self, *args, **kwargs):
        super(MainWindow, self).__init__(*args, **kwargs)
        self.graphWidget = pg.PlotWidget()
        self.buttonWidget = QtWidgets.QPushButton()

        self.buttonWidget.clicked.connect(self.button_clicked)

        layout = QtWidgets.QVBoxLayout()
        layout.addWidget(self.graphWidget)
        layout.addWidget(self.buttonWidget)
        widget = QtWidgets.QWidget()
        widget.setLayout(layout)
        self.setCentralWidget(widget)

        rng = np.random.default_rng()
        length = 10_000
        self.y_data_a = rng.normal( scale=0.1, size=length )
        self.y_data_b = rng.normal( scale=0.1, size=length )
        self.y_data_c = rng.normal( scale=0.1, size=length )
        self.y_data_d = rng.normal( scale=0.1, size=length )
        idx = rng.integers( length, size=3)
        err = np.abs( rng.normal( scale=2e10, size=3 ) )
        self.y_data_b[ idx ] += err
        self.y_data_c[ idx ] -= 0.000001*err
        self.y_data_d[ idx ] -= err
        self.x_data = [idx for idx, val in enumerate(self.y_data_a)]
        self.graphWidget.setYRange( -0.1, 0.1)
        self.item0 = pg.PlotDataItem( x=self.x_data, y=self.y_data_a, pen='g')
        self.item1 = pg.PlotDataItem( x=self.x_data, y=self.y_data_b, pen='r')
        self.item2 = pg.PlotDataItem( x=self.x_data, y=self.y_data_c, pen='m')

        disable = False
        for item in [self.item0, self.item1, self.item2]:
            self.graphWidget.addItem( item )
            if disable: item.setDynamicRangeLimit(None)

    def button_clicked(self):
        self.item2.setData(x=self.x_data, y=self.y_data_d)

def main():
    app = QtWidgets.QApplication(sys.argv)
    main = MainWindow()
    main.show()
    sys.exit(app.exec_())

if __name__ == '__main__':
    main()
```

### Tested environment

 * Qt Python binding: PySide2 5.15.2 Qt 5.15.2
 * Python version: 3.9.1
 * NumPy version: 1.19.4
 * Operating system: macOS Catalina 10.15.7
 * Installation method: pip

(Though I suspect this issue is environment-independent.)

**Comment by EfremBraun** [comment]
@NilsNemitz @ j9ac9k
Figure you guys want to know.

**Comment by NilsNemitz** [comment]
Dear @EfremBraun , thank you very much for the report. The example was very helpful!

I hope the revised code of PR #1691 will address this, and make sure the the dynamic range limiter also runs immediately after setting fresh data.

I have to admit that on some level it's nice to see that I am not the only one with misbehaving data sets that keep triggering the "disappearing curves" Qt problem. :)

**Comment by EfremBraun** [comment]
I can confirm that PR #1691 fixes this issue.

**Comment by j9ac9k** [comment]
Fixed with merging #1691
